### DISCUSSION TITLE
2D Mesh Cutting

### DISCUSSION THREAD

**Discussion by 19kdc3** [marcomusy/vedo#798] [Q&A] [answered]
Hi Marco,

I have 2, 2D meshes in which I'd like to cut one out of the other. 

In this example below, I'd like to cut the blue mesh out of the red mesh. Thus, the result would be the square red mesh with a smaller square hole in it where the blue square mesh was placed.
![image](https://user-images.githubusercontent.com/73371848/215189077-37e91503-c790-4f6c-b0b7-7ba7823e7232.png)

I've tried using the `boolean`, `cut_with_mesh`, and `cut_with_point_loop` tools, but all fall short in various ways. 

Do you have any suggestions for how I could achieve my goal? Thanks!

**Comment by marcomusy** [comment, marked as answer]
Hi
the fact that they are red and blue tells me you have already a cell array associted to the Mesh.
You can use then `.threshold(..., on="cells")`:
https://vedo.embl.es/docs/vedo.html?search=threshold

**Reply by 19kdc3** [reply]
Oh sorry, I should have made it more clear. That image is simply the 2 meshes superimposed on each other.

**Reply by marcomusy** [reply]
Uhm to be honest I could not think of a better way of doing it - maybe you can for advice in the vtk discourse forum... maybe someone has some clever ideas..

**Reply by 19kdc3** [reply]
Would potentially either of the following be possible/seem feasible?

1. Converting to another object type, doing the "boolean" operation, then converting back
2. "merging" the meshes together, then "boolean"

**Reply by marcomusy** [reply]
Refer to #805
